src/herder/HerderImpl.h

```diff
@@ -356,5 +367,18 @@ class HerderImpl : public Herder
     ConsensusData mTrackingSCP;
 
     uint32_t mMaxTxSize{0};
+
+    // Key: {ledger state hash, tx hash, lower bound close time, upper bound
```

Comment:
I don't mind including the ledger close bounds here, but since the validity status may fluctuate from ledger to ledger, I'm not sure if this is too useful compared to the additional complexity this introduces.

Reply:
No, we have to include close time offsets. Otherwise we could potentially use cached validity results that are incorrect. E.g. imagine we validate a transaction that comes via the mempool and cache the result as valid. When the next leader proposes the block, it has to validate the transaction against the new proposed close time. This should cause a cache miss, and re-validation of the same tx against the new time bounds. Impact of re-using invalid cache results can be catastrophic, as nodes on the network won't be able to deterministically validate blocks.